bundles/org.openhab.binding.caddx/src/main/java/org/openhab/binding/caddx/internal/CaddxCommunicator.java

```diff
@@ -0,0 +1,521 @@
+                            "CaddxCommunicator.run() Received packet checksum does not match. in: {} {}, calc {} {}",
+                            incomingMessage.getChecksum1In(), incomingMessage.getChecksum2In(),
+                            incomingMessage.getChecksum1Calc(), incomingMessage.getChecksum2Calc());
+                }
+            }
+        } catch (IOException e) {
+            logger.debug("CaddxCommunicator.run() IOException. Stopping sender thread. {}", getSerialPortName());
+            Thread.currentThread().interrupt();
+        }
+
+        logger.warn("CaddxCommunicator.run() Sender thread stopped. {}", getSerialPortName());
+    }
+
+    /**
+     * Event handler to receive the data from the serial port
+     *
+     * @param SerialPortEvent serialPortEvent The event that occurred on the serial port
+     */
+    @Override
+    public void serialEvent(@Nullable SerialPortEvent serialPortEvent) {
+        if (serialPortEvent == null) {
+            return;
+        }
+
+        if (serialPortEvent.getEventType() == SerialPortEvent.DATA_AVAILABLE) {
+            logger.trace("Data receiving from the serial port");
+            if (protocol == CaddxProtocol.Binary) {
+                receiveInBinaryProtocol(serialPortEvent);
+            } else {
+                receiveInAsciiProtocol(serialPortEvent);
+            }
+        }
+    }
+
+    private int readByte(InputStream stream) {
+        try {
+            return stream.read();
+        } catch (IOException e) {
+            return -1;
+        }
+    }
+
+    private void receiveInBinaryProtocol(SerialPortEvent serialPortEvent) {
+        int b = 0;
+
+        // Read the start byte
+        if (!inMessage) // skip until 0x7E
+        {
+            b = 0;
+            while (b != 0x7E && b != -1) {
+                b = readByte(in);
+            }
+            if (b == -1) {
+                return;
+            }
+
+            inMessage = true;
+            messageBufferLength = 0;
+        }
+        logger.trace("CaddxCommunicator.handleBinaryProtocol() Got start byte");
+
+        // Read the message length
+        if (messageBufferLength == 0) {
+            b = readByte(in);
+            if (b == -1) {
+                return;
+            }
+
+            messageBufferLength = b + 2; // add two bytes for the checksum
+            message = new byte[messageBufferLength];
+        }
+        logger.trace("CaddxCommunicator.handleBinaryProtocol() Got message length {}", b);
+
+        // Read the message
+        while (true) {
+            b = readByte(in);
+            if (b == -1) {
+                return;
+            }
+            message[messageBufferIndex] = (byte) b;
+
+            if (message[messageBufferIndex] == 0x7D) {
+                unStuff = true;
+                continue;
+            }
+
+            if (unStuff) {
+                message[messageBufferIndex] |= 0x20;
+                unStuff = false;
+            }
```

Comment:
This `unStuff` field is only used here, so can you make it a local variable?

Reply:
No. It needs to be a field to handle a possible EOF condition between the bytes.